libpod/options.go

```diff
@@ -1831,6 +1832,31 @@ func WithHostDevice(dev []specs.LinuxDevice) CtrCreateOption {
 	}
 }
 
+func WithDeviceWeights(weights map[string]specs.LinuxWeightDevice) CtrCreateOption {
```

Comment:
This isn't necessary, you can directly alter the OCI spec in https://github.com/containers/podman/blob/main/pkg/specgen/generate/oci.go#L177 which returns the final spec that is used by Libpod. Moving this code to there will remove the need for Libpod changes.